lib/ecto_diff.ex

```diff
@@ -367,7 +367,47 @@ defmodule EctoDiff do
     end
   end
 
-  defp diff_association(previous, current, %{cardinality: :many, field: field, related: struct}, acc, opts) do
+  defp diff_association(
+         previous,
+         current,
+         %{cardinality: :many, through: [through, assoc_field], owner: owner},
+         acc,
+         opts
+       ) do
+    assoc_diff =
+      acc
+      |> Keyword.get_values(through)
+      |> List.flatten()
+      |> Enum.map(& &1[:changes][assoc_field])
+
+    diff =
+      case assoc_diff do
+        [_ | _] = diff ->
+          List.flatten(diff)
+
+        _ ->
```

Comment:
:information_source: This should just be an empty list:
```suggestion
        [] ->
```

Reply:
At the time, the return could also be `[nil]`.  Changed the way we handle this.